tests/gocase/unit/introspection/introspection_test.go

```diff
@@ -20,33 +20,227 @@
+			r := c.Do(ctx, "CLIENT", "SETNAME", "foobar")
+			require.NoError(t, r.Err())
+			require.Equal(t, "OK", r.Val())
+			require.Regexp(t, ".*name=foobar.*", rdb.ClientList(ctx).Val())
+		}()
+
+		// now the client should no longer be listed
+		require.Eventually(t, func() bool {
+			r := rdb.ClientList(ctx).Val()
+			return !strings.Contains(r, "foobar")
+		}, 5*time.Second, 100*time.Millisecond)
+	})
+
+	t.Run("Kill normal client", func(t *testing.T) {
+		defer func() { rdb = srv.NewClient() }()
+
+		c := srv.NewClient()
+		defer func() { require.NoError(t, c.Close()) }()
+		r := c.Do(ctx, "CLIENT", "SETNAME", "normal")
+		require.NoError(t, r.Err())
+		require.Equal(t, "OK", r.Val())
+		require.Regexp(t, ".*name=normal.*", rdb.ClientList(ctx).Val())
+
+		require.EqualValues(t, 1, rdb.ClientKillByFilter(ctx, "skipme", "yes", "type", "normal").Val())
+		require.EqualValues(t, 1, rdb.ClientKillByFilter(ctx, "skipme", "no", "type", "normal").Val())
+
+		// reconnect
+		rdb = srv.NewClient()
+		defer func() { require.NoError(t, rdb.Close()) }()
+
+		// now the client should no longer be listed
+		require.Eventually(t, func() bool {
+			r := rdb.ClientList(ctx).Val()
+			return !strings.Contains(r, "normal")
+		}, 5*time.Second, 100*time.Millisecond)
+	})
+
+	t.Run("Kill pubsub client", func(t *testing.T) {
+		// subscribe clients
+		c0 := srv.NewClient()
+		defer func() { require.NoError(t, c0.Close()) }()
+		require.NoError(t, c0.Do(ctx, "CLIENT", "SETNAME", "pubsub").Err())
+		r := c0.Do(ctx, "SUBSCRIBE", "foo")
+		require.NoError(t, r.Err())
+		require.Equal(t, "[subscribe foo 1]", fmt.Sprintf("%v", r.Val()))
+
+		// psubscribe clients
+		c1 := srv.NewClient()
+		defer func() { require.NoError(t, c1.Close()) }()
+		require.NoError(t, c1.Do(ctx, "CLIENT", "SETNAME", "pubsub").Err())
+		r = c1.Do(ctx, "PSUBSCRIBE", "bar.*")
+		require.NoError(t, r.Err())
+		require.Equal(t, "[psubscribe bar.* 1]", fmt.Sprintf("%v", r.Val()))
+
+		// normal clients
+		c2 := srv.NewClient()
+		require.NoError(t, c2.Do(ctx, "CLIENT", "SETNAME", "normal").Err())
+		defer func() { require.NoError(t, c2.Close()) }()
+
+		require.EqualValues(t, 2, rdb.ClientKillByFilter(ctx, "type", "pubsub").Val())
+
+		// now the pubsub client should no longer be listed
+		// but normal client should not be dropped
+		require.Eventually(t, func() bool {
+			r := rdb.ClientList(ctx).Val()
+			fmt.Printf("CLIENT LIST: %v\n", r)
+			return strings.Count(r, "pubsub") == 0 && strings.Count(r, "normal") == 1
+		}, 5*time.Second, 100*time.Millisecond)
+	})
+
+	t.Run("DEBUG will freeze server", func(t *testing.T) {
+		// use TCPClient to avoid waiting for reply
+		c := srv.NewTCPClient()
+		defer func() { require.NoError(t, c.Close()) }()
+		require.NoError(t, c.WriteArgs("DEBUG", "sleep", "2.2"))
+		
+	    // Sleep 100ms to prevent the successive set command to be executed
+		// before the debug command since there are in the different connection.
+		time.Sleep(100 * time.Millisecond)
```

Comment:
```suggestion
        // sleep 100ms to prevent the successive set command to be executed
        // before the debug command since there are in the different connection.
		time.Sleep(100 * time.Millisecond)
```